### ISSUE TITLE
Country code syntax error

### ISSUE THREAD

**Issue by jamespblloyd** [worthbak/apple-store-inventory-checker#5]
Hello, I tried out your lovely script but I got an error message that appears to be related to the country code argument, but I think that I am adding it correctly (store ID is Australian and I add AU to the end of my command). 

`$ node index.js R386 AU`
```
/home/jamesl/bin/apple-store-inventory-checker-main/index.js:15
  const passedCountry = args[1] ?? "US";
                                 ^

SyntaxError: Unexpected token ?
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
```

**Comment by worthbak** [comment]
some changes were just merged in that addressed a few known issues with Australian usage (#4) - wanna give this a shot again and let me know if those changes resolved your issue?

**Comment by Shamrock13** [comment]
Having same issue for US based check

**Comment by mikejk8s** [comment]
I'm having this same issue with US stores now on WSL2. This runs fine on my macbook which is interesting. Same revision (current latest), problems started after last night. I'm digging around trying to fix it I'll report back if I do. @jamespblloyd what are you running this on?

**Comment by srikat** [comment]
Just want to report that

```
node index.js R504 AU
```

works fine here. Tested about an hour ago.

**Comment by jamespblloyd** [comment]
> I'm having this same issue with US stores now on WSL2. This runs fine on my macbook which is interesting. Same revision (current latest), problems started after last night. I'm digging around trying to fix it I'll report back if I do. @jamespblloyd what are you running this on?

I am also running it on WSL (I have tried it on both WSL1 and 2 now and both give me the same error message). I don't see how using WSL should cause a problem but it seems to be....all the more reason for me to buy a mac ¯\_(ツ)_/¯

**Comment by Shamrock13** [comment]
Works fine on WSL searching for AU store. 

What versions are you guys running?

**Comment by worthbak** [comment]
interesting - tough issue. I'm unable to reproduce the issue on my local, though I'm on macOS 12.0.1, so that's not particularly surprising. the error, `SyntaxError: Unexpected token ?`, is interesting - I can't tell if that's `node` complaining about actually invalid syntax, or if the contents of `args` are unexpected. 

would someone mind logging the contents of `process.argv` and reporting what they find? should just need to add this to line 6 of `index.js`:

```js
console.log(process.argv);
```

**Comment by mikejk8s** [comment]
Pulled latest (w/ german SKUs), added console.log to #6 -- same output though. I'm on a completely fresh WSL2 install w/ just nodejs installed, tbh I haven't done much if any node in WSL, just golang so might be some weird WSLism, not a big deal for  me since I've got a work MBP sitting next to it

```
apple-inventory-checker$ node index.js  R047 
/home/mrj/workspace/redo/apple-store-inventory-checker/index.js:42
  country = (args[1] ?? "US").toUpperCase();
                      ^

SyntaxError: Unexpected token ?
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
```

apple-store-inventory-checker$ node --version
v10.19.0

**Comment by worthbak** [comment]
hm, the `node --version` is substantially different from mine - I'm on `v17.0.1`. would WSL have a different version structure for node? seems odd.

**Comment by mrcasual** [comment]
TL;DR: You need Node ≥15 to run this script.

1. Nullish coalescing operator (`??`) is [only supported ](https://node.green/#ES2020-features--nullish-coaescing-operator-----) in Node ≥14.

2. [String.prototype.replaceAll](https://github.com/worthbak/apple-store-inventory-checker/blob/d478225a1f942ecd71826885485df43fa0edf320/index.js#L117) is [only supported](https://node.green/#ES2021-features--String-prototype-replaceAll) in Node ≥15.

**Comment by jyeung79** [comment]
> TL;DR: You need Node ≥15 to run this script.
> 
> 1. Nullish coalescing operator (`??`) is [only supported ](https://node.green/#ES2020-features--nullish-coaescing-operator-----) in Node ≥14.
> 2. [String.prototype.replaceAll](https://github.com/worthbak/apple-store-inventory-checker/blob/d478225a1f942ecd71826885485df43fa0edf320/index.js#L117) is [only supported](https://node.green/#ES2021-features--String-prototype-replaceAll) in Node ≥15.

Yep this is correct. You'll need at least version 15. I recommend gettting `nvm` so you can manage multiple node versions and switch between them.

**Comment by worthbak** [comment]
ah, there it is - thanks y'all. I also recommend `nvm` 👍 weird that @mikejk8s's fresh install of node gave them such an outdated version? anyway, would love to see one of y'all use a more recent (`>=15`) version of node and retry!

**Comment by Shamrock13** [comment]
same thing happened to me @worthbak. I had to manually install the newest version. MS must be using some old linux that's not up-to-date.

**Comment by worthbak** [comment]
odd choice by MSFT (if that's indeed what's going on). in any case, @jamespblloyd given that this is your issue, if you want to try the proposed solution of updating your version of `node`, we can close out this issue?

**Comment by mrcasual** [comment]
> same thing happened to me @worthbak. I had to manually install the newest version. MS must be using some old linux that's not up-to-date.

Just to be fair to MSFT, they offer a number of distros, including Ubuntu 20.04 LTS (the latest LTS version) that ships with Node 10.19, which is what @mikejk8s is using.

[This PR](https://github.com/worthbak/apple-store-inventory-checker/pull/6) should fix it.

**Comment by worthbak** [comment]
@mrcasual haha fair - didn't mean to rag on Microsoft, it's awesome how much work they've put into WSL 👍 thanks for the PR! I'll take a look later this evening.

**Comment by mikejk8s** [comment]
Thanks I didn't think to check how old that was at all!

**Comment by dfrencham** [comment]
I'm not sure catering to old versions of node is the way to go usually, but for this script it is probably the best option given the broad usage. 

I can run the script in both WSL and cmd.exe. Assuming you are using Ubuntu under WSL, node is pretty easy to update: https://joshtronic.com/2021/05/09/how-to-install-nodejs-16-on-ubuntu-2004-lts/
